### PR TITLE
[MRG]:whale: Update modelci docker image

### LINKED ISSUE #284: Failed to convert model to ONNX format
#### Software and Hardware Versions 
ModelCI v1.x.x, CUDA Version vx.x.x, GPU device used...

#### Problem description

When register a model by model API with convert=True options, the model failed to be converted into ONNX format

#### Steps to Reproduce the Problem
image

#### Expected Behavior
Model can be registered successfully without any error

#### Other Information 
Things you tried, stack traces, related issues, suggestions on how to fix it...

```
INFO:     127.0.0.1:55044 - "POST /api/v1/model/?convert=true&profile=false HTTP/1.1" 500 Internal Server Error
ERROR:    Exception in ASGI application
Traceback (most recent call last):
  File "~/miniconda3/envs/test/lib/python3.7/site-packages/uvicorn/protocols/http/h11_impl.py", line 396, in run_asgi
    result = await app(self.scope, self.receive, self.send)
  File "~/miniconda3/envs/test/lib/python3.7/site-packages/uvicorn/middleware/proxy_headers.py", line 45, in __call__
    return await self.app(scope, receive, send)
  File "~/miniconda3/envs/test/lib/python3.7/site-packages/fastapi/applications.py", line 179, in __call__
    await super().__call__(scope, receive, send)
  File "~/miniconda3/envs/test/lib/python3.7/site-packages/starlette/applications.py", line 111, in __call__
    await self.middleware_stack(scope, receive, send)
  File "~/miniconda3/envs/test/lib/python3.7/site-packages/starlette/middleware/errors.py", line 181, in __call__
    raise exc from None
  File "~/miniconda3/envs/test/lib/python3.7/site-packages/starlette/middleware/errors.py", line 159, in __call__
    await self.app(scope, receive, _send)
  File "~/miniconda3/envs/test/lib/python3.7/site-packages/starlette/middleware/cors.py", line 86, in __call__
    await self.simple_response(scope, receive, send, request_headers=headers)
  File "~/miniconda3/envs/test/lib/python3.7/site-packages/starlette/middleware/cors.py", line 142, in simple_response
    await self.app(scope, receive, send)
  File "~/miniconda3/envs/test/lib/python3.7/site-packages/starlette/exceptions.py", line 82, in __call__
    raise exc from None
  File "~/miniconda3/envs/test/lib/python3.7/site-packages/starlette/exceptions.py", line 71, in __call__
    await self.app(scope, receive, sender)
  File "~/miniconda3/envs/test/lib/python3.7/site-packages/starlette/routing.py", line 566, in __call__
    await route.handle(scope, receive, send)
  File "~/miniconda3/envs/test/lib/python3.7/site-packages/starlette/routing.py", line 227, in handle
    await self.app(scope, receive, send)
  File "~/miniconda3/envs/test/lib/python3.7/site-packages/starlette/routing.py", line 41, in app
    response = await func(request)
  File "~/miniconda3/envs/test/lib/python3.7/site-packages/fastapi/routing.py", line 183, in app
    dependant=dependant, values=values, is_coroutine=is_coroutine
  File "~/miniconda3/envs/test/lib/python3.7/site-packages/fastapi/routing.py", line 133, in run_endpoint_function
    return await dependant.call(**values)
  File "~/ML-Model-CI/modelci/app/v1/endpoints/model.py", line 138, in publish_model
    models = register_model(model=model, convert=convert, profile=profile)
  File "~/ML-Model-CI/modelci/hub/manager.py", line 75, in register_model
    model_dir_list.extend(_generate_model_family(model))
  File "~/ML-Model-CI/modelci/hub/manager.py", line 178, in _generate_model_family
    if converter.convert(net, 'pytorch', 'onnx', save_path=onnx_dir, inputs=inputs, outputs=outputs, model_input=model_input, optimize=False):
  File "~/ML-Model-CI/modelci/hub/converter/converter.py", line 38, in convert
    return converter(model, **kwargs)
  File "~/ML-Model-CI/modelci/hub/converter/to_onnx.py", line 131, in from_pytorch
    dtype = model_data_type_to_torch(input_.dtype)
  File "~/ML-Model-CI/modelci/types/type_conversion.py", line 127, in model_data_type_to_torch
    f'model_dtype is expecting one of the type: `int`, `str`, or `DataType` but got {type(model_dtype)}'
TypeError: model_dtype is expecting one of the type: `int`, `str`, or `DataType` but got <enum 'DataType'>

```

replace this line:

https://github.com/cap-ntu/ML-Model-CI/blob/d20f6d76a1fd98ba5a8e9f8cd3908c2a951eec3c/modelci/types/type_conversion.py#L133

with:

```
from modelci.types.models.common import DataType
```

this problem gone

### PR DESCRIPTION
**Pull request by univerone**
Closes #284

#### What has been done to verify that this works as intended?

- Build three docker images
  - cpu version(`mlmodelci/mlmodelci:cpu`) 
  - cuda10.2+cudnn8 version(`mlmodelci/mlmodelci:cuda10.2-cudnn8`)
  - frontend image `mlmodelci/frontend`
- Add two docker compose file to start basic modelhub services(To be implemented in later work)
- Update readme 

Start modelci basic modelhub service(CPU version) by the following command:
```
docker-compose -f ML-Model-CI/docker/docker-compose-cpu-modelhub.yml up -d
```
image

After all services started, you can go to <http://localhost:3333/#/> to browse frontend page

You can run cli command via container, like:
```bash
docker exec -it modelci-backend modelci modelhub ls
```
image

#### Before submitting this PR, please make sure you have:

- [x] run `python -m pytest tests/` and confirmed all checks still pass.
- [x] verified that any code or assets from external sources are properly credited.

### CODE REVIEW

File: docker/docker-compose-gpu-modelhub.yml [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+    restart: always
+    ports:
+      - 27017:27017
+    environment:
+      MONGO_INITDB_USERNAME: modelci
+      MONGO_INITDB_PASSWORD: modelci@2020
+      MONGO_INITDB_DATABASE: modelci
+    hostname: mongodb
+    volumes:
+      - ../scripts/init_db.js:/docker-entrypoint-initdb.d/init_db.js
```
Comment (YuanmingLeee) [inline docker/docker-compose-gpu-modelhub.yml:18]: better to use absolute path

File: docker/gpu.Dockerfile [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+RUN python3.7 -m venv $VIRTUAL_ENV
+ENV PATH="$VIRTUAL_ENV/bin:$PATH"
+
+COPY . /content
+# Build tvm
+WORKDIR /root
+RUN git clone https://github.com/apache/tvm tvm --recursive
+
+WORKDIR /root/tvm
+RUN mkdir -p build \
```
Comment (YuanmingLeee) [inline docker/gpu.Dockerfile:38]: Better to use
```sugguestion
RUN cd /root/tvm \
 && mkdir -p build \
```

File: setup.py [hunk 3]
```diff
@@ -114,7 +114,7 @@ def install_triton_client():
     cuda_version_file = Path(f'{os.environ["CUDA_HOME"]}/version.txt')
     if cuda_version_file.exists():
         with open(cuda_version_file) as f:
-            CUDA_VERSION = "cu".join(f.readline().strip().split(" ")[-1].split(".")[:2])
+            CUDA_VERSION = "cu"+"".join(f.readline().strip().split(" ")[-1].split(".")[:2])
```
Comment (YuanmingLeee) [inline setup.py:117]: space between 'cu' + ''

File: docker/gpu.Dockerfile [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+RUN apt-get update \
+ && apt-get install llvm-10 \
+ libopenblas-dev \
+ lsof \
+ libgl1-mesa-glx \
+ libglib2.0-0 \
+ python${PYTHON_VER}-distutils \
+ python${PYTHON_VER} \
+ python${PYTHON_VER}-venv \
+ python${PYTHON_VER}-dev -y\
```
Comment (YuanmingLeee) [inline docker/gpu.Dockerfile:87]: Do not need a new line among installed libraries

File: docker/gpu.Dockerfile [hunk 5]
```diff
@@ -0,0 +1,95 @@
+# Stage1: Compile
```
Comment (YuanmingLeee) [inline docker/gpu.Dockerfile:1]: better to name as cuda10.2

File: README.md [hunk 6]
```diff
@@ -47,79 +47,143 @@ Several features are in beta testing and will be available in the next release s
...
+![CLI start service](https://i.loli.net/2021/04/12/wtyMuIlaJdsVchX.gif)
+
+Or stop the service by:
+
+```bash
+modelci service stop
 ```
 
-### Docker
+![CLI stop service](https://i.loli.net/2021/04/12/Hdi4JTclKkEaLfh.gif)
```
Comment (YuanmingLeee) [inline README.md:90]: Why your characters have space in between? Is this a font encoding problem?

File: docker/gpu.Dockerfile (hunk 5 above)
Comment (HuaizhengZhang) [reply docker/gpu.Dockerfile:1]: agree. gpu as the name is too general

File: README.md [hunk 7]
```diff
@@ -47,79 +47,143 @@ Several features are in beta testing and will be available in the next release s
...
+
+
+## Installation Guide
+
+### Prerequisites
+
+- A GNU/Linux environment(Ubuntu preferred)
+- [Docker](https://docs.docker.com/engine/install/)
+- [Docker Compose](https://docs.docker.com/compose/) (for Installation via Docker)
+- [TVM](https://github.com/apache/incubator-tvm) and `tvm` Python module
```
Comment (HuaizhengZhang) [inline README.md:65]: I suppose TVM and TensorRT should be optional software. We do not require user to install it for enjoying modelci

File: README.md [hunk 8]
```diff
@@ -47,79 +47,143 @@ Several features are in beta testing and will be available in the next release s
...
+```bash
+modelci service stop
 ```
 
-### Docker
+![CLI stop service](https://i.loli.net/2021/04/12/Hdi4JTclKkEaLfh.gif)
+
+### Installation using Docker
 
 ![](https://img.shields.io/docker/pulls/mlmodelci/mlmodelci.svg) ![](https://img.shields.io/docker/image-size/mlmodelci/mlmodelci)
```
Comment (HuaizhengZhang) [inline README.md:94]: Better update the badge
One for CPU and one for GPU

File: setup.py [hunk 9]
```diff
@@ -169,6 +169,6 @@ def install_triton_client():
     DEPENDENCY_LINKS=[TORCH_INSTALL_URL, TORCHVISION_INSTALL_URL],
     entry_points='''
         [console_scripts]
-        modelci=modelci.cli:cli
+        modelci=modelci.cli:typer_click_object
```
Comment (HuaizhengZhang) [inline setup.py:172]: has this been well verified?

**Review by HuaizhengZhang — CHANGES_REQUESTED** [review]

File: setup.py (hunk 9 above)
Comment (univerone) [reply setup.py:172]: yes,  the entrypoint for ModelCI CLI interface has been changed to `typer_click_object` function, `cli` function is no longer avaliable, https://github.com/cap-ntu/ML-Model-CI/blob/8ba8605b3bf8547ec37f6ad002c7706c7620e1b2/modelci/cli/__init__.py#L41-L42

File: README.md (hunk 6 above)
Comment (univerone) [reply README.md:90]: It's related to the font, I'll use monospaced font and replace all the images.

File: docker/gpu.Dockerfile (hunk 2 above)
Comment (univerone) [reply docker/gpu.Dockerfile:38]: It's recommended to use WORKDIR refer to  https://github.com/hadolint/hadolint/wiki/DL3003

File: docker/gpu.Dockerfile (hunk 4 above)
Comment (univerone) [reply docker/gpu.Dockerfile:87]: Maybe it's more clear, I just refer to https://docs.docker.com/develop/develop-images/dockerfile_best-practices/#run, I'll add the version for each package to obey the lint rules

**Review by HuaizhengZhang — APPROVED** [review]

**Pull request merged**